src/coreclr/jit/gentree.cpp

```diff
@@ -4889,6 +4889,32 @@ bool Compiler::gtMarkAddrMode(GenTree* addr, int* pCostEx, int* pCostSz, var_typ
     return false;
 }
 
+//------------------------------------------------------------------------------
+// gtGetFlagsForOperand : Get flags for an operation over the operant
+//
+// Arguments:
+//    oper   - Performed operation
+//    op     - Operand
+//
+// Return Value:
+//    Flags required by the operand
+GenTreeFlags Compiler::gtGetFlagsForOperand(genTreeOps oper, GenTree* op)
+{
+    GenTreeFlags flags = GTF_EMPTY;
+    if ((oper == GT_IND) && op->IsIconHandle())
+    {
+        flags |= GTF_IND_NONFAULTING;
+
+        GenTreeFlags handleKind = op->GetIconHandleFlag();
+        if ((handleKind != GTF_ICON_STATIC_HDL) && (handleKind != GTF_ICON_BBC_PTR) &&
+            (handleKind != GTF_ICON_GLOBAL_PTR))
+        {
+            flags |= GTF_IND_INVARIANT;
+        }
+    }
```

Comment:
This invariantness setting is fragile; it does not check what type is being loaded.

Reply:
Yes, I had the same concern - like why do we expect invariantess if we acess a byte location with long type